src/middlewared/middlewared/plugins/filesystem.py

```diff
@@ -52,6 +62,24 @@ def resolve_cluster_path(self, path, ignore_ctdb=False):
         cluster_path = path.replace(FuseConfig.FUSE_PATH_SUBST.value, f'{FuseConfig.FUSE_PATH_BASE.value}/')
         return cluster_path
 
+    @accepts(Str('path'))
+    @returns()
+    def remove_directory(self, path):
+        """
+        Remove a directory at the specified path.
+        """
+        p = pathlib.Path(path)
+        if not p.is_absolute():
+            raise CallError(f'{path}: not an absolute path.', errno.EINVAL)
+
+        if not p.exists():
+            raise CallError(f'{path}: path does not exist.', errno.ENOENT)
+
+        if not os.path.realpath(path).startswith('/mnt/'):
+            raise CallError(f'{path}: path not permitted', errno.EPERM)
+
+        shutil.rmtree(path)
```

Comment:
This seems like a rather sharp knife. Is this only for doing regression tests? If so, why not just use a SSH_TEST?